S7.Net/PLC.cs

```diff
@@ -107,11 +107,35 @@ public bool IsConnected
             {
                 try
                 {
+                    bool clientConnected = false;
+
                     if (tcpClient == null)
                         return false;
 
                     //TODO: Actually check communication by sending an empty TPDU
-                    return tcpClient.Connected;
+                    if (tcpClient.Client.Poll(0,SelectMode.SelectRead))
```

Comment:
Not sure if this tells us anything useful, true is also returned on closed sockets.